koap/src/jsMain/kotlin/Hex.kt

```diff
@@ -0,0 +1,26 @@
+package com.juul.koap
+
+internal fun parseHexBinary(hex: String): ByteArray {
+    val len = hex.length
+    require(len % 2 == 0) { "Hex binary string must be even length" }
+    val bytes = ByteArray(len / 2)
+    var i = 0
+
+    while (i < len) {
+        val h = hexToInt(hex[i])
+        val l = hexToInt(hex[i + 1])
+        require(!(h == -1 || l == -1)) { "Invalid hex chars: ${hex[i]}${hex[i + 1]}" }
+
+        bytes[i / 2] = ((h shl 4) + l).toByte()
+        i += 2
+    }
+
+    return bytes
+}
+
+private fun hexToInt(character: Char): Int = when (character) {
+    in '0'..'9' -> character - '0'
+    in 'A'..'F' -> character - 'A' + 10
+    in 'a'..'f' -> character - 'a' + 10
+    else -> -1
+}
```

Comment:
I feel like this could be dramatically simplified by using [String.toByte](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-byte.html), especially in conjunction with [CharSequence.chunked](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/chunked.html)

Reply:
Perhaps those aren't multiplatform? Or there's some compatibility issue?
I only say that because this chunk of code is a direct copy from `kotlinx.serialization`.

Reply:
https://github.com/Kotlin/kotlinx.serialization/blob/56d2b5043d323b058fc3fd64baf44f4c00c4808b/formats/cbor/commonTest/src/kotlinx/serialization/HexConverter.kt#L10-L33

Reply:
That being said, we should probably give attribution (and have a comment in the file saying where it came from).

Reply:
Added comments to the functions stating where they came from.